### PR TITLE
Add checkbox for notification settings UI

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

- new feature toggle `profileUseNotificationSettingsCheckboxes` added to platform (vets-api PR for toggle also in review)
- using the toggle, a `va-checkbox` will render for each notification item channel (email or text) instead of the existing radio buttons
- creates the support for text and/or email to be supported for each item
- all legacy radio button functionality is maintained, and will remain on production while this feature set is being developed
- wires up the core functionality of the checkbox
  - first pass at hooking the value change listener to doing the api call to set the preference
  - sets the checked value based on the isAllowed value that comes from the communication_preferences api data
  - DOES NOT wire up the default send indicator _yet_ but does add the `defaultSendIndicator` to the redux state so that it can be used as this feature work moves forward.
  - sets up saving, success, and error messages for each checkbox during api call lifecycle

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/59915

## Testing done

- All existing tests passing and locally tested radio buttons for regressions and also tested checkboxes displaying via toggle.
- Added base unit test for notification group to test checkbox display logic via feature toggle
- Will be adding various unit and e2e tests as this work progresses. This first PR is to get the toggle wired up and foundation built.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| | Screenshot 2023-06-20 at 12 39 01 PM | Screenshot 2023-06-20 at 12 38 29 PM |

## What areas of the site does it impact?

Profile - Notification Settings | For staging only

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution

### CODE REVIEW

**Comment by adamwhitlock1** [comment]
Working on figuring out why the ci/cd e2e tests are failing on one unrelated test.

File: src/applications/personalization/profile/components/notification-settings/NotificationChannelCheckboxesFieldset.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        <legend className="rb-legend vads-u-padding--0">
+          <h3 className={legendClasses}>{itemName}</h3>
+        </legend>
+      </div>
+      {description ? (
+        <p className="vads-u-margin-y--0p5 vads-u-color--gray-medium">
+          {description}
+        </p>
+      ) : null}
+      {/* {!loadingMessage && !successMessage && !warningMessage && errorSpan}
```
Comment (allisonlu) [inline src/applications/personalization/profile/components/notification-settings/NotificationChannelCheckboxesFieldset.jsx:30]: I see this in `NotificationCheckbox.jsx`, can you delete these conditionals from this file?

File: src/applications/personalization/profile/components/notification-settings/NotificationRadioButtons.jsx [hunk 2]
```diff
@@ -85,13 +58,17 @@ const NotificationRadioButtons = ({
   if (errorMessage) {
     errorSpanId = `${id}-error-message`;
     errorSpan = (
-      <MessageWrapper id={errorSpanId} classes="rb-input-message-error" alert>
+      <NotificationStatusMessage
```
Comment (allisonlu) [inline src/applications/personalization/profile/components/notification-settings/NotificationRadioButtons.jsx:61]: yay we love semantic component names 🫡

**Review by allisonlu — APPROVED** [review]
everything else lgtm!

File: src/applications/personalization/profile/components/notification-settings/NotificationChannelCheckboxesFieldset.jsx (hunk 1 above)
Comment (adamwhitlock1) [reply src/applications/personalization/profile/components/notification-settings/NotificationChannelCheckboxesFieldset.jsx]: Good call. Fixed.

**Review by rsmithadhoc — APPROVED** [review]

**Pull request merged**